### PR TITLE
Fix iframe reloaded on each theme change

### PR DESCRIPTION
**Pull request by witoszekdev**
I want to merge this change because it fixes an issue when changing Dashboard's theme: it caused app's iframe to reload, even though the theme in app changed correctly (thanks to AppBridge - postMessage)

**PR intended to be tested with API branch:** 

### Screenshots

Before:

https://user-images.githubusercontent.com/4144459/231511102-477719ae-5d8b-471b-9e8f-64cdd4826337.mp4

After:

https://user-images.githubusercontent.com/4144459/231511341-fb8d1125-2de3-46ae-bbb7-d84d2ff43e01.mp4

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: src/apps/components/AppFrame/AppIFrame.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    // Otherwise this will cause reload of entire iframe
+    useEffect(() => {
+      themeRef.current = themeType;
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    const iframeSrc = AppUrls.resolveAppIframeUrl(appId, src, {
+      ...params,
+      featureFlags,
+      theme: themeRef.current,
```
Comment (krzysztofzuraw) [inline src/apps/components/AppFrame/AppIFrame.tsx:32]: **question**: if we rely on events do you need to pass the theme here?

**Review by krzysztofzuraw — COMMENTED** [review]
Ping @lkostrowski for visibility 😀

File: src/apps/components/AppFrame/AppFrame.tsx
Comment (lkostrowski) [inline src/apps/components/AppFrame/AppFrame.tsx:1]: Maybe we should rename this file now? We have `AppFrame` and `AppIFrame`. I believe original `Frame` was also referring to the <iframe>

File: src/apps/components/AppFrame/AppIFrame.tsx (hunk 1 above)
Comment (lkostrowski) [reply src/apps/components/AppFrame/AppIFrame.tsx:32]: @krzysztofzuraw yes, app can read theme before AppBridge is mounted and set theme early

File: src/apps/components/AppFrame/AppIFrame.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+);
+
+export const AppIFrame = memo(_AppIFrame, (prev, next) => {
+  const res = isEqualWith(prev, next, (prevVal, nextVal) => {
+    // don't compare functions
+    if (typeof prevVal === "function" && typeof nextVal === "function") {
+      return true;
+    }
+  });
+  return res;
```
Comment (lkostrowski) [inline src/apps/components/AppFrame/AppIFrame.tsx:55]: I think you can just return L49

File: src/apps/components/AppFrame/AppIFrame.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        className={className}
+        sandbox="allow-same-origin allow-forms allow-scripts"
+      />
+    );
+  },
+);
+
+export const AppIFrame = memo(_AppIFrame, (prev, next) => {
+  const res = isEqualWith(prev, next, (prevVal, nextVal) => {
+    // don't compare functions
```
Comment (lkostrowski) [inline src/apps/components/AppFrame/AppIFrame.tsx:50]: why? Shouldnt functions be wrapped with useCallback (so they have same ref?)

**Review by lkostrowski — APPROVED** [review]
Minor comments, no blockers.

**Review by poulch — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Pull request merged**